app/workers/evss/disability_compensation_form/job_status.rb

```diff
@@ -1,198 +1,15 @@
-            updated_at: timestamp
-          }
-
-          form_job_status = Form526JobStatus.find_by(job_id: job_id)
-          bgjob_errors = form_job_status.bgjob_errors || {}
-          new_error = {
-            "#{timestamp.to_i}": {
-              caller_method: __method__.to_s,
-              error_class: error_class,
-              error_message: error_message,
-              timestamp: timestamp
-            }
-          }
-          bgjob_errors.merge!(new_error)
-          values[:bgjob_errors] = bgjob_errors
-
-          Form526JobStatus.upsert(values, unique_by: :job_id)
-
-          Rails.logger.error(
-            'Form526 Exhausted', submission_id: msg['args'].first,
-                                 job_id: job_id,
-                                 error_class: error_class,
-                                 error_message: error_message
-          )
-          Metrics.new(statsd_key_prefix).increment_exhausted
-        rescue => e
-          Rails.logger.error('error tracking job exhausted', error: e, class: msg['class'].demodulize)
-        end
-        # rubocop:enable Metrics/MethodLength
-      end
-
-      # Code wrapped by this block will run between the {job_try} and {job_success} methods
-      #
-      # @param job_title [String] Description of the job being run
-      # @param saved_claim_id [Integer] The {SavedClaim} id
-      # @param submission_id [Integer] The {Form526Submission} id
-      #
-      def with_tracking(job_title, saved_claim_id, submission_id, is_bdd = nil)
-        @status_job_title = job_title
-        @status_saved_claim_id = saved_claim_id
-        @status_submission_id = submission_id
-        @is_bdd = is_bdd
-
-        job_try
-        yield
-        job_success
-      end
-
-      # Metrics and logging for each Sidekiq try
-      #
-      def job_try
-        upsert_job_status(Form526JobStatus::STATUS[:try])
-        log_info('try')
-        metrics.increment_try
-      rescue => e
-        Rails.logger.error('error tracking job try', error: e, class: klass)
-      end
-
-      # Metrics and logging for when the job succeeds
-      #
-      def job_success
-        upsert_job_status(Form526JobStatus::STATUS[:success])
-        log_info('success')
-        metrics.increment_success(@is_bdd)
-      rescue => e
-        Rails.logger.error('error tracking job success', error: e, class: klass)
-      end
-
-      # Metrics and logging for any retryable errors that occurred.
-      # Retryable are system recoverable, e.g. an upstream http timeout
-      #
-      def retryable_error_handler(error)
-        upsert_job_status(Form526JobStatus::STATUS[:retryable_error], error)
-        log_error('retryable_error', error)
-        metrics.increment_retryable(error, @is_bdd)
-      end
-
-      # Metrics and logging for any non-retryable errors that occurred.
-      # Non-retryable errors will always fail, e.g. an ArgumentError in the job class
+      include Sidekiq::Form526JobStatusTracker::JobTracker
```

Comment:
Could we remove this evss-specific module and the evss metrics module under this directory entirely if we just included those modules in EVSS::DisabilityCompensationForm::Job?

Reply:
I _think_ I tried that and things broke elsewhere that I wasn't prepared to address. Let me try once more just to be sure though because that would definitely be cleaner.